java.lang.String cannot be converted to okhttp3.MediaType

### Check Ahead

- [x] I have searched the [issues](https://github.com/seata/seata/issues) of this repository and believe that this is not a duplicate.

- [ ] I am willing to try to fix this bug myself.


### Ⅰ. Issue Description

https://github.com/apache/incubator-seata/actions/runs/20753876763/job/59590388727
```
Error:  Failed to execute goal org.apache.maven.plugins:maven-compiler-plugin:3.8.1:compile (default-compile) on project seata-common: Compilation failure: Compilation failure: 
Error:  /Users/runner/work/incubator-seata/incubator-seata/common/src/main/java/org/apache/seata/common/util/HttpClientUtil.java:[108,30] no suitable method found for create(java.lang.String,okhttp3.MediaType)
Error:      method okhttp3.RequestBody.create(okhttp3.MediaType,java.lang.String) is not applicable
Error:        (argument mismatch; java.lang.String cannot be converted to okhttp3.MediaType)
Error:      method okhttp3.RequestBody.create(okhttp3.MediaType,okio.ByteString) is not applicable
Error:        (argument mismatch; java.lang.String cannot be converted to okhttp3.MediaType)
Error:      method okhttp3.RequestBody.create(okhttp3.MediaType,byte[]) is not applicable
Error:        (argument mismatch; java.lang.String cannot be converted to okhttp3.MediaType)
Error:      method okhttp3.RequestBody.create(okhttp3.MediaType,java.io.File) is not applicable
Error:        (argument mismatch; java.lang.String cannot be converted to okhttp3.MediaType)
```
I suspect that different versions of Spring Boot depend on different versions of OkHttp, which could cause compatibility issues. To avoid conflicts arising from changes in Spring Boot versions, we should manually specify a fixed, reliable version of OkHttp 3.

### Ⅱ. Describe what happened

_No response_

### Ⅲ. Describe what you expected to happen

_No response_

### Ⅳ. How to reproduce it (as minimally and precisely as possible)

_No response_

### Ⅴ. Anything else we need to know?

_No response_

### Ⅵ. Environment

_No response_

@funky-eyes 
Hi. I was just about to share my progress.

### Test Environment

I selected one of the failing CI configurations to reproduce the issue:
```bash
./mvnw -T 4C clean test -P args-for-client-test \
  -Dspring-boot.version=2.3.12.RELEASE \
  -Dspring-framework.version=5.2.15.RELEASE \
  -e -B
```

---

### Step 1: Added OkHttp 4.9.3 → Apollo Test Failed

**Action:** Added `<okhttp.version>4.9.3</okhttp.version>` to `seata-dependencies/pom.xml`

**Result:**
```
[ERROR] org.apache.seata.config.apollo.ApolloConfigurationTest
java.lang.NoSuchMethodError: 'void okhttp3.internal.Internal.initializeInstanceForTests()'
```

**Root Cause:** Apollo Client requires OkHttp 3.x internally, but project now uses 4.x globally.

---

### Step 2: Fixed Apollo Module → Console Module Failed

**Action:** Isolated Apollo test with OkHttp 3.14.9 in `config/seata-config-apollo/pom.xml`
```xml
<dependency>
    <groupId>com.squareup.okhttp3</groupId>
    <artifactId>okhttp</artifactId>
    <version>3.14.9</version>
    <scope>test</scope>
</dependency>
```

**Result:** Apollo tests pass ✅, but new errors in console module:
```
[ERROR] cannot find symbol: class HttpStatusCode (Spring 6.0+ only)
[ERROR] cannot access javax.servlet.Filter
[ERROR] incompatible types: JwtAuthenticationTokenFilter cannot be converted to jakarta.servlet.Filter
```

**Root Cause:** Console module uses Spring Boot 3.x APIs, incompatible with Spring Boot 2.x tests in CI.

---

This is where I am now. Could I get some feedback on how to proceed?

> [@funky-eyes](https://github.com/funky-eyes) Hi. I was just about to share my progress.
> 
> ### Test Environment
> I selected one of the failing CI configurations to reproduce the issue:
> 
> ./mvnw -T 4C clean test -P args-for-client-test \
>   -Dspring-boot.version=2.3.12.RELEASE \
>   -Dspring-framework.version=5.2.15.RELEASE \
>   -e -B
> ### Step 1: Added OkHttp 4.9.3 → Apollo Test Failed
> **Action:** Added `<okhttp.version>4.9.3</okhttp.version>` to `seata-dependencies/pom.xml`
> 
> **Result:**
> 
> ```
> [ERROR] org.apache.seata.config.apollo.ApolloConfigurationTest
> java.lang.NoSuchMethodError: 'void okhttp3.internal.Internal.initializeInstanceForTests()'
> ```
> 
> **Root Cause:** Apollo Client requires OkHttp 3.x internally, but project now uses 4.x globally.
> 
> ### Step 2: Fixed Apollo Module → Console Module Failed
> **Action:** Isolated Apollo test with OkHttp 3.14.9 in `config/seata-config-apollo/pom.xml`
> 
> <dependency>
>     <groupId>com.squareup.okhttp3</groupId>
>     <artifactId>okhttp</artifactId>
>     <version>3.14.9</version>
>     <scope>test</scope>
> </dependency>
> **Result:** Apollo tests pass ✅, but new errors in console module:
> 
> ```
> [ERROR] cannot find symbol: class HttpStatusCode (Spring 6.0+ only)
> [ERROR] cannot access javax.servlet.Filter
> [ERROR] incompatible types: JwtAuthenticationTokenFilter cannot be converted to jakarta.servlet.Filter
> ```
> 
> **Root Cause:** Console module uses Spring Boot 3.x APIs, incompatible with Spring Boot 2.x tests in CI.
> 
> This is where I am now. Could I get some feedback on how to proceed?

It looks like the issue has now been resolved — thank you!
I’d like to know which version of Apollo Client supports OkHttp 4.x. If we could switch to a compatible version, that would be ideal.
Regarding the JwtAuthenticationTokenFilter issue you mentioned earlier, would you mind creating a new issue for it?
Additionally, I believe we should enforce that the console and namingserver modules must be compiled with Spring Boot 3 and JDK 25. When building with Spring Boot 2, these two modules should not be compiled or tested at all.

--- a/changes/en-us/2.x.md
+++ b/changes/en-us/2.x.md
@@ -32,6 +32,7 @@ Add changes here for all PR submitted to the 2.x branch.
 ### optimize:
 
 - [[#7930](https://github.com/apache/incubator-seata/pull/7930)] pin the Spring version for namingserver and console
+- [[#7934](https://github.com/apache/incubator-seata/pull/7934)] add OkHttp and MockWebServer dependencies to resolve version conflicts
 
 ### security:
 
@@ -54,6 +55,7 @@ Thanks to these contributors for their code commits. Please report an unintended
 - [LegendPei](https://github.com/LegendPei)
 - [funky-eyes](https://github.com/funky-eyes)
 - [maple525866](https://github.com/maple525866)
+- [neronsoda](https://github.com/neronsoda)
 
 
 Also, we receive many valuable issues, questions and advices from our community. Thanks for you all.

--- a/changes/zh-cn/2.x.md
+++ b/changes/zh-cn/2.x.md
@@ -32,6 +32,7 @@
 ### optimize:
 
 - [[#7930](https://github.com/apache/incubator-seata/pull/7930)] 固定namingserver和console的Spring版本
+- [[#7934](https://github.com/apache/incubator-seata/pull/7934)] 添加 OkHttp 和 MockWebServer 依赖来解决版本冲突
 
 
 ### security:
@@ -55,6 +56,7 @@
 - [LegendPei](https://github.com/LegendPei)
 - [funky-eyes](https://github.com/funky-eyes)
 - [maple525866](https://github.com/maple525866)
+- [neronsoda](https://github.com/neronsoda)
 
 
 同时，我们收到了社区反馈的很多有价值的issue和建议，非常感谢大家。

--- a/dependencies/pom.xml
+++ b/dependencies/pom.xml
@@ -145,6 +145,8 @@
 
         <!--  for fory  -->
         <fory.version>0.12.3</fory.version>
+
+        <okhttp.version>4.9.3</okhttp.version>
     </properties>
 
     <dependencyManagement>
@@ -904,8 +906,8 @@
                 <groupId>commons-io</groupId>
                 <artifactId>commons-io</artifactId>
                 <version>${commons-io.version}</version>
-             </dependency>
-                <dependency>
+            </dependency>
+            <dependency>
                 <groupId>com.fasterxml.jackson.core</groupId>
                 <artifactId>jackson-databind</artifactId>
                 <version>${jackson.version}</version>
@@ -929,6 +931,16 @@
                 <groupId>com.shentongdata</groupId>
                 <artifactId>oscarJDBC8</artifactId>
                 <version>${oscar.version}</version>
+            </dependency>
+            <dependency>
+                <groupId>com.squareup.okhttp3</groupId>
+                <artifactId>okhttp</artifactId>
+                <version>${okhttp.version}</version>
+            </dependency>
+            <dependency>
+                <groupId>com.squareup.okhttp3</groupId>
+                <artifactId>mockwebserver</artifactId>
+                <version>${mockwebserver-version}</version>
             </dependency>
         </dependencies>
     </dependencyManagement>

